pkg/cluster/kubeclient/backup.go

```diff
@@ -22,7 +22,7 @@ func (u *Kubeclientset) BackupCluster(ctx context.Context, backupName string) er
 
 	job := &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      backupName,
+			Name:      "etcd-backup",
```

Comment:
Why hardcode this constant? I see it's also used above on line 18... was this a bug?

Reply:
Yeah I think this should be reverted as well.

Reply:
(waiting on an answer to this question, otherwise PR looks good to me)

Reply:
So the validation on a kubernetes name is much more rigorous than a blob - to the extent that we can't use the timestamp (that is at the end of the backup name). Since the job is a transient entity, there is no real reason to have the job and blob named the same. Also since the blob lives on, i'd contend that it's name is more important.

Reply:
I think its fine. It will create something like `etcd-backup-d8cc6`. And as stated blob-name is versioned.